sklearn/ensemble/tests/test_forest.py

```diff
@@ -1153,38 +1153,6 @@ def test_class_weight_errors(name):
     check_class_weight_errors(name)
 
 
-def check_warm_start(name, random_state=42):
-    # Test if fitting incrementally with warm start gives a forest of the
-    # right size and the same results as a normal fit.
-    X, y = hastie_X, hastie_y
-    ForestEstimator = FOREST_ESTIMATORS[name]
-    est_ws = None
-    for n_estimators in [5, 10]:
-        if est_ws is None:
-            est_ws = ForestEstimator(n_estimators=n_estimators,
-                                     random_state=random_state,
```

Comment:
Change this to `np.random.RandomState(42)` here and below for `est_no_ws`, and the test will fail. It obviously fail too if you pass `None`.

So the artificial `random_state` consumption as done in `fit` above is not able to properly enforce the intended behavior anyway.